pkg/collector/corechecks/ebpf/probe/ebpfcheck/probe.go

```diff
@@ -1192,3 +1197,40 @@ func hashMapNumberOfEntriesWithHelper(mp *ebpf.Map, mapid ebpf.MapID, mphCache *
 
 	return int64(res), nil
 }
+
+// https://lwn.net/ml/linux-mm/20230202014158.19616-6-laoar.shao@gmail.com/
+// https://github.com/torvalds/linux/commit/304849a27b341cf22d5c15096144cc8c1b69e0c0
+func preciseMapMemUsageSupported() bool {
+	kv, err := kernel.HostVersion()
+	if err != nil {
+		return false
+	}
+
+	return kv >= kernel.VersionCode(6, 4, 0)
+}
+
+// https://elixir.bootlin.com/linux/v6.4/source/kernel/bpf/syscall.c#L804
+func readPreciseMapMemUsageFromFdInfo(fd int) uint64 {
```

Comment:
Lets return an explicit error here if anything fails instead of returning zero and going to the fallback. This will help us address any wrong assumptions in the future more easily.